### PR TITLE
Drop Gaffer Configuration from Gretty

### LINKED ISSUE #249: Remove Groovy-based Logging Configuration from Gretty
Gretty uses the `GafferConfigurator`, which configures Logback using a gretty-supplied Groovy script.

Logback removed the ability to use Groovy for configuring logging in both versions, 1.2.9 onwards, and 1.3.0-alpha4 onwards: https://logback.qos.ch/news.html

The article states:

> In response to [CVE-2021-42550](https://cve.report/CVE-2021-42550) (aka [LOGBACK-1591](https://jira.qos.ch/browse/LOGBACK-1591)) we have decided to make the following steps.
>
>[...]
>
>4) Removed Groovy configuration support. As logging is so pervasive and configuration with Groovy is probably too powerful, this feature is unlikely to be reinstated for security reasons.

TL;DR.: using Groovy to configure Logback is gone, and that way of configuring Logback won't return. We'll need to remove Gretty's dependency on Gaffer in eventually both, 3.x and 4.x.

### PR DESCRIPTION
**Pull request by f4lco**
🚧 ...fixes #249... 🚧

### CODE REVIEW

**Comment by f4lco** [comment]
@boris-petrov please give this a look. I think the most significant change is that with this PR, we also discontinue support for webapps Groovy script for configuring Logback, not only for Gretty. In my opinion, there is no point in dragging on support for Logback Groovy configuration scripts for the webapps. If Gretty benefits from the mitigation of [CVE-2021-42550](https://cve.report/CVE-2021-42550), webapps should benefit in the same way going forward. If the app is old, the app can use old versions of Gretty, both being affected by vulns in dated Logback.

**Comment by boris-petrov** [comment]
This PR is amazing, you've put a lot of thought and effort on it, thank you!

The `Port * to XML` commits I only glanced over so let's hope there isn't some mistake there but it's not that important really. :smile: 

I wanted to understand why the [first commit](https://github.com/gretty-gradle-plugin/gretty/pull/254/commits/0734e91ece4b6db24a809d9bad7e4fc24de4d36f) does all the configuration in code rather than using something like a template (similar to the [last commit](https://github.com/gretty-gradle-plugin/gretty/pull/254/commits/b1ccfcacc5be177c831346f2024fbb1864cb912d))? What am I missing here?

**Comment by f4lco** [comment]
> I wanted to understand why the [first commit](https://github.com/gretty-gradle-plugin/gretty/pull/254/commits/0734e91ece4b6db24a809d9bad7e4fc24de4d36f) does all the configuration in code rather than using something like a template (similar to the [last commit](https://github.com/gretty-gradle-plugin/gretty/pull/254/commits/b1ccfcacc5be177c831346f2024fbb1864cb912d))? What am I missing here?

That's a really good question. I think the failure to plan ahead shines through - I did not know beforehand we got this templated log configuration in Gretty.
So firstly, I "simply" wanted to inline the Logback configuration script into Gretty, because Logback dropped support for Groovy configuration scripts. Using the programmatic configuration of Logback seemed the obvious choice.
Later in the process, I encountered the templated Logback configuration of Gretty products, and needed to touch and refactor the template. Up to now, it did not occur to me both implementations work in eerily similar ways.

So the way I look at it now, we have a programmatic configuration of Logback, which looks relatively clean, and for which we can rely on a compiler to check if we got the calling conventions correct. And then the template for XML configuration exists, which I think is a necessary evil. Because the template is error-prone and less maintainable, I needed to write additional tests to get even my implementation correct. I think the little-known templating language and XML hodgepodge plus the additional escaping rules (like `\$`) force an additional burden on the maintainers.

In summary, I'm not sure if converging on one implementation is the correct thing to do here, which would be the XML template. Perhaps Gretty's logging configuration and the logging configuration of Gretty products are simply distinct entities, and ought to live independently. I'm afraid one day we'll end up with a supercharged, do-it-all type of template with weird bugs, which contains both, switches for Gretty's internal logging configuration, as well as for the apps 🤷 
Anyway, glad to exchange opinions on this topic!

**Comment by f4lco** [comment]
> The Port * to XML commits I only glanced over so let's hope there isn't some mistake there but it's not that important really. 😄

If you have seen one instance of the XML configuration, you have seen all! It's pretty much the same log configuration in all integration tests, except that the name of the logfile will match the name of the containing webapp project, e.g., project "websocket" logs into "websocket.log".

**Comment by boris-petrov** [comment]
Thanks for the explanation! I would in principal prefer to use the XML approach because that's the "standard" way as far as I know - i.e. Joran is well supported and documented. Also, it's much less code. Of course, the downside is the lack of compiler support (and writing XML which is bad in itself) but oh well. :) I'm fine either way so if you prefer leaving the code as-is, I'm totally OK! I'll leave the decision to you.

Otherwise great work as usual! The beers keep on accumulating! :)

Merge at will when you decide on the comment above. :)

**Comment by f4lco** [comment]
Thanks @boris-petrov! Yes, I think I'll leave it that, I see too little benefit in shifting things around, also wrt the dependency graph. Likely we are better off to keep "Gretty the gradle plugin packaging 'products'" and "Gretty the runner process running webapps" as distinct as humanly possible.

**Comment by boris-petrov** [comment]
@f4lco - I guess we should also push these changes to `gretty-3.x`? Should I or would you prefer doing it?

**Comment by f4lco** [comment]
@boris-petrov  let me try 😄
